src/model.rs

```diff
@@ -257,24 +278,44 @@ impl Component for Model {
             Action::ChangeInput(coord, new_value) => {
                 let old_grammar = self.grammars.get_mut(&coord);
                 match old_grammar {
-                    Some(g @ Grammar { kind: Kind::Text(_), .. }) => {
+                    Some(
+                        g @ Grammar {
+                            kind: Kind::Text(_),
+                            ..
+                        },
+                    ) => {
                         self.console.log(&new_value);
                         g.kind = Kind::Text(new_value);
-                    },
-                    _ => ()
+                    }
+                    _ => (),
                 }
                 false
             }
 
-            Action::ShowSuggestions(coord, query) => {
-                false
-            }
+            Action::ShowSuggestions(coord, query) => false,
 
             Action::SetActiveCell(coord) => {
+                self.f_select_cell = coord.clone();
+                info!(
+                    "f-row-{} f-column-{}",
+                    self.f_select_cell.row_to_string(),
+                    self.f_select_cell.col_to_string()
+                );
                 self.active_cell = Some(coord);
                 true
             }
 
+            Action::SetSelectedCells(coord) => {
+                self.l_select_cell = coord.clone();
+                info!(
+                    "l-row-{} l-column-{}",
+                    self.l_select_cell.row_to_string(),
+                    self.l_select_cell.col_to_string()
+                );
+                self.select_grammar.push(coord);
```

Comment:
i think what you want to do here is _add_ all the grammars between `f_select_cell` and `l_select_cell` to `self.select_grammar`. you'll need to loop through the coordinates somehow. one thing you might want to check for is that `f_select_cell` and `l_select_cell` have the same parent.

Reply:
yeah. It's just some testing for that. Got it. Thank man